app/view_models/paye/PayeYourTaxableIncome.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package view_models.paye
+
+import models.{DataHolder, PayeAtsData}
+import view_models.Amount
+
+case class PayeYourTaxableIncome(
+                                  taxYear: Int,
+                                  incomeTaxRows: List[IncomeTaxRow],
+                                  incomeBeforeTaxTaxTotal: Amount
+                                ) extends TaxYearFormatting {
+}
+
+object PayeYourTaxableIncome {
+  def buildViewModel(payeAtsData: PayeAtsData): Option[PayeYourTaxableIncome] = {
+    val taxRows = getIncomeTaxRows(payeAtsData.income_data).filter(row => row.value.nonZero)
+
+    Some(PayeYourTaxableIncome(
```

Comment:
When can this be None? Why is it optional?

Reply:
Agreed, this should just construct the view model

Done 👍